### PR TITLE
Feature/create route from proxy response 139

### LINKED ISSUE #139: Create routes from proxy responses
When I'm mocking a api, firstly I run my app proxied and so I copy the data from the log to a new route.

It can be helpful a button on the route of the environment log to use that request/response as a mock.

### PR DESCRIPTION
**Pull request by gabrielperes97**
**Description**

I create a button on a log item for a proxied request. This button create a route using that log response/request. 

**Related Issue**

Issue #139

**Motivation and Context**

Described in #139 

**How Has This Been Tested?**

I create two environments, the first with the "GET /answer" route and the second without routes but with the proxy enabled  pointed to the first. I made a "GET /answer" request to the second environment and looked at the logs, clicked on the Mock button, restarted the environment and made the request again. Now the "GET /anwser" response is mocked.

**Screenshots**

Screenshot_20191019_174026

Screenshot_20191019_174117

**Closing issues**

Closes #139

### CODE REVIEW

File: test/proxy.spec.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+describe('Proxy', () => {
+  tests.runHooks();
+
+  it('Start environments', async () => {
+    await tests.helpers.startEnvironment();
+    await tests.helpers.selectEnvironment(2);
+    await tests.helpers.startEnvironment();
+  });
+
+  it ('Call /anwser', () => {
```
Comment (255kb) [inline test/proxy.spec.ts]: please use async / await to avoid race conditions

File: src/app/services/environments.service.ts [hunk 2]
```diff
@@ -643,4 +643,44 @@ export class EnvironmentsService {
...
     this.eventsService.injectHeaders.emit({ target: 'environmentHeaders', headers: CORSHeaders });
   }
+
+  /**
+   * Create a route based on a environment log entry
+   */
+  public createRouteFromLog(logUUID?: string) {
+    const environmentsLogs = this.store.get('environmentsLogs');
+    const uuidEnvironment = this.store.get('activeEnvironmentUUID');
+    const log = environmentsLogs[uuidEnvironment].find(environmentLog => environmentLog.uuid === logUUID)
```
Comment (255kb) [inline src/app/services/environments.service.ts]: add missing semicolon (please run the linter before pushing :) )

**Comment by 255kb** [comment]
@gabrielperes97 This is a very high quality contribution! Thank you. 
I requested two minor changes. 

I also have a question: why limit this feature only to proxied request? I know that some people would be highly interested to create mocks from any entering requests (caught or not, proxied or not). 
Could you modify your code so it allows mock creation for non-caught request?

Thanks!

File: test/proxy.spec.ts (hunk 1 above)
Comment (gabrielperes97) [reply test/proxy.spec.ts]: Done.

File: src/app/services/environments.service.ts (hunk 2 above)
Comment (gabrielperes97) [reply src/app/services/environments.service.ts]: I really forgot to run the linter, sorry. But it's fixed.

**Comment by gabrielperes97** [comment]
> @gabrielperes97 This is a very high quality contribution! Thank you.
> I requested two minor changes.
> 
> I also have a question: why limit this feature only to proxied request? I know that some people would be highly interested to create mocks from any entering requests (caught or not, proxied or not).
> Could you modify your code so it allows mock creation for non-caught request?
> 
> Thanks!

You're right, it's really useful. I've modified to mock any log entry.

**Comment by 255kb** [comment]
Thank you @gabrielperes97 for this high quality contribution!

**Pull request merged**